`ld x86_64 error` under mac os mojave (10.14+) with darwin kernel 18+

For macos mojave, compile failed for ld (x86_64) error.
Under mac os mojave (10.14+) with darwin kernel 18+.

```
Undefined symbols for architecture x86_64:
  "_IRConst_U32", referenced from:
      _zero_division_side_exits in postprocess.o
  "_IRConst_U64", referenced from:
      _zero_division_side_exits in postprocess.o
  "_IRExpr_Binop", referenced from:
      _zero_division_side_exits in postprocess.o
  "_IRExpr_Const", referenced from:
      _mips32_post_processor_fix_unconditional_exit in postprocess.o
      _zero_division_side_exits in postprocess.o
  "_IRExpr_RdTmp", referenced from:
      _zero_division_side_exits in postprocess.o
  "_IRStmt_Exit", referenced from:
      _zero_division_side_exits in postprocess.o
  "_IRStmt_WrTmp", referenced from:
      _zero_division_side_exits in postprocess.o
  "_LibVEX_Init", referenced from:
      _vex_init in pyvex.o
  "_LibVEX_Lift", referenced from:
      _vex_lift in pyvex.o
  "_LibVEX_Update_Control", referenced from:
      _vex_lift in pyvex.o
  "_LibVEX_default_VexAbiInfo", referenced from:
      _vex_init in pyvex.o
  "_LibVEX_default_VexArchInfo", referenced from:
      _vex_init in pyvex.o
  "_LibVEX_default_VexControl", referenced from:
      _vex_init in pyvex.o
  "_addStmtToIRSB", referenced from:
      _irsb_insert in postprocess.o
      _zero_division_side_exits in postprocess.o
  "_newIRTemp", referenced from:
      _zero_division_side_exits in postprocess.o
  "_sizeofIRType", referenced from:
      _collect_data_references in analysis.o
  "_typeOfIRExpr", referenced from:
      _get_default_exit_target in analysis.o
      _collect_data_references in analysis.o
      _zero_division_side_exits in postprocess.o
  "_typeOfIRTemp", referenced from:
      _collect_data_references in analysis.o
ld: symbol(s) not found for architecture x86_64
```

--- a/Makefile-gcc
+++ b/Makefile-gcc
@@ -20,6 +20,13 @@ CC_NATIVE ?= cc
 AR ?= ar
 RM ?= rm -f
 
+# Don't worry about Windows not having shell, Makefile-msvc will deal
+# with it`
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S), Darwin)
+	AR = /usr/bin/ar 
+endif
+
 STATIC_LIBRARY_FILE = libvex.a
 DYNAMIC_LIBRARY_FILE = libvex.so
 EXTRA_CLEAN_FILES = TAG-* auxprogs/genoffsets

